time_execution/backends/threaded.py

```diff
@@ -74,12 +75,17 @@ def send_metrics():
                 logger.warning('%r write failure %r', self.backend, exc)
 
         while self.has_work():
+            if self.thread:
```

Comment:
One of the advantages I see in switching to a non-daemonic thread is that you can gracefully terminate its tasks. But you're terminating the thread even if there are tasks in the queue.